src/cli/compiler.js

```diff
@@ -66,6 +67,102 @@ class Compiler {
     }
   }
 
+  getFilesToWatch() {
+    try {
+      let allFiles = []
+      let subgraph = this.loadSubgraph()
+
+      // Add all file paths specified in manifest
+      allFiles.push(subgraph.getIn(['schema', 'file']))
+      subgraph.get('dataSources').map(dataSource => {
+          allFiles.push(dataSource.getIn(['mapping', 'file']))
+          dataSource.getIn(['mapping', 'abis']).map(abi => {
+            allFiles.push(abi.get('file'))
+          })
+      })
+
+      // Make paths absolute
+      let allAbsolutePaths = allFiles.map(file => {
+        return path.resolve(this.sourceDir, file)
+      })
+      return allAbsolutePaths
+    } catch(e) {
+      this.logger.fatal('Failed to parse subgraph file locations:', e)
+    }
+  }
+
+  watchAndCompile() {
+    let compiler = this
+    compiler.logger.info('')
+
+    // Initialize watcher
+    let watcher = chokidar.watch(this.options.subgraphManifest, {
+      persistent: true,
+      ignoreInitial: true,
+      atomic: 500
+    })
+
+    // Get locations of all files in subgraph manifest
+    let watchedFiles = this.getFilesToWatch()
+    watcher.add(watchedFiles)
+
+    // Add event listeners
+    watcher
+      .on('ready', function() {
+        compiler.logger.info(chalk.grey("Watching relevant manifest files"))
+        compiler.compile()
+      })
+      .on('change', path => {
+        compiler.logger.info('%s %s',
+          chalk.grey('File change detected: '),
+          compiler.displayPath(path)
+        )
+        compiler.logger.currentStep = 0
+        if (path === compiler.options.subgraphManifest) {
```

Comment:
Can you double-check also that these two have any chance of being equal? I'm assuming `path` will be absolute — is it guaranteed that `subgraphManifest` is? If not, we need to make it so that they can be compared.